### PR TITLE
Prevent possible XSS attack vector in the input fields of the group section

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Calls the already existing `utils.escapeHtml()` to make sure nothing is executed.  Works independent of CSP changes

### CODE REVIEW

File: scripts/pi-hole/js/utils.js [hunk 1]
```diff
@@ -13,11 +13,10 @@ function escapeHtml(text) {
     "&": "&amp;",
     "<": "&lt;",
     ">": "&gt;",
-    '"': "&quot;",
```
Comment (XhmikosR) [inline scripts/pi-hole/js/utils.js]: Not sure about this removal. The `escapeHtml` function is used in plenty other places.

Comment (PromoFaux) [reply scripts/pi-hole/js/utils.js]: I have another thought actually, we can leave the `"` in this function and adjust the php function in `groups.php` to replace `&quot;` with `"`, that should have the same effect. It's just that the groups function requires it.

**Comment by XhmikosR** [comment]
You need to include `utils.js` in `dns_records.php` before `ip-address-sorting.js`.

I still need to test this myself before approving.

**Comment by XhmikosR** [comment]
I only tested group-domains.php and it seems to work. Someone should try the other pages just to be sure.

**Review by XhmikosR — DISMISSED** [review]

File: scripts/pi-hole/php/groups.php [hunk 2]
```diff
@@ -58,7 +58,10 @@ function JSON_error($message = null)
 } elseif ($_POST['action'] == 'add_group') {
     // Add new group
     try {
-        $names = str_getcsv(trim($_POST['name']), ' ');
+        // Payload from the web interface will have been sanitised with " converted to &quot;
+        // so before we process the input, we will do the opposite and replace &quot; with ";
+        $input = str_replace("&quot;","\"",trim($_POST['name']));
```
Comment (XhmikosR) [inline scripts/pi-hole/php/groups.php]: Just to confirm, shouldn't it make sense to use something like https://www.php.net/manual/en/function.html-entity-decode.php?

Comment (PromoFaux) [reply scripts/pi-hole/php/groups.php]: Fair shout, I didn't know of it's existence before now. Either or works in this case.

**Comment by XhmikosR** [comment]
We should prefer the built in solution assuming it works the same IMHO
because it's tested and works for everything, not just quotes.

**Comment by XhmikosR** [comment]
I'd like @DL6ER to confirm the database-related code. For example I see this from the [PHP page](https://www.php.net/manual/en/function.html-entity-decode.php):

>Although this argument is technically optional, you are highly encouraged to specify the correct value for your code if you are using PHP 5.5 or earlier, or if your [default_charset](https://www.php.net/manual/en/ini.core.php#ini.default-charset) configuration option may be set incorrectly for the given input. 

I will try the branch again tomorrow, and try to review all JS code in those files.

**Comment by PromoFaux** [comment]
1c5bbc1f92c85ccd465969985172ce3d4b8db3c0 prevents an (unlikely, but possible) attack vector:

Given the following entry in the static lease file:

```
dhcp-host=AA:BB:CC:DD:EF:FF,192.168.1.1,<script>alert("hello")</script>
```

Before patch:

image

image

After patch script is no longer run:

image

**Comment by DL6ER** [comment]
@XhmikosR I cannot test anything until beginning/mid of next week. I fully trust @PromoFaux to be aware of any database-related issues that might be there. Note that it is planed to drop any SQL stuff from the web interface for Pi-hole v6.0 and to shift any list manipulations into the API (provided by FTL). This to reduce the number of places where stuff can get changed and where write-access is needed.

**Comment by PromoFaux** [comment]
There are a couple more places I want to check before merging (see my screenshots on mattermost for more info)

**Review by DL6ER — APPROVED** [review]
Tested and found no oddities.

**Comment by PromoFaux** [comment]
Thanks to Dino at @telspacesystems and also @spartacvs for reporting the issues that lead to this PR

**Comment by iasdeoupxe** [comment]
CVE-2020-14971 seems got assigned for this issue:

https://blog.telspace.co.za/2020/06/pi-hole-code-injection-cve-2020-14971.html

**Pull request merged**